RetailCoder.VBE/Inspections/QuickFixes/AssignedByValParameterMakeLocalCopyQuickFix.cs

```diff
@@ -0,0 +1,182 @@
+﻿using Rubberduck.Inspections.Abstract;
+using System.Linq;
+using Rubberduck.Parsing;
+using Rubberduck.VBEditor;
+using Rubberduck.Inspections.Resources;
+using Rubberduck.Parsing.Grammar;
+using Rubberduck.Parsing.Symbols;
+using System.Windows.Forms;
+using Rubberduck.UI.Refactorings;
+using Rubberduck.Common;
+using Antlr4.Runtime;
+using System.Collections.Generic;
+using Antlr4.Runtime.Tree;
+
+namespace Rubberduck.Inspections.QuickFixes
+{
+    public class AssignedByValParameterMakeLocalCopyQuickFix : QuickFixBase
+    {
+        private readonly Declaration _target;
+        private string _localCopyVariableName;
+        private string[] _variableNamesAccessibleToProcedureContext;
+        private bool _isQuickFixUnitTest;
+
+        public AssignedByValParameterMakeLocalCopyQuickFix(Declaration target, QualifiedSelection selection)
+            : base(target.Context, selection, InspectionsUI.AssignedByValParameterMakeLocalCopyQuickFix)
+        {
+            _target = target;
+            _isQuickFixUnitTest = false;
+            _localCopyVariableName = "x" + _target.IdentifierName.CapitalizeFirstLetter();
+            _variableNamesAccessibleToProcedureContext = GetVariableNamesAccessibleToProcedureContext(_target.Context.Parent.Parent);
+        }
+
+        public override bool CanFixInModule { get { return false; } }
+        public override bool CanFixInProject { get { return false; } }
+
+        //This function exists solely to support unit testing 
+        public void TESTONLY_FixUsingAutoGeneratedName(string suggestedName = "")
```

Comment:
I don't like this... a class shouldn't have the knowledge of whether or not it's being executed in a test context. There *has* to be a better way.